pkg/plugin/conntrack/_cprog/conntrack.c

```diff
@@ -155,40 +167,55 @@ static __always_inline bool _ct_handle_udp_connection(struct packet *p, struct c
  * @arg observation_point The point in the network stack where the packet is observed.
  */
 static __always_inline bool _ct_handle_tcp_connection(struct packet *p, struct ct_v4_key key, struct ct_v4_key reverse_key, __u8 observation_point) {
-    // Check if the packet is a SYN packet.
-    if (p->flags & TCP_SYN) {
-        // Update packet accordingly.
-        p->is_reply = false;
-        p->traffic_direction = _ct_get_traffic_direction(observation_point);
-        // Create a new connection with a timeout of CT_SYN_TIMEOUT.
-        return _ct_create_new_tcp_connection(key, p->flags, observation_point);
-    }
+    struct ct_entry *entry = bpf_map_lookup_elem(&retina_conntrack, &key);
+    struct ct_entry *reverse_entry = bpf_map_lookup_elem(&retina_conntrack, &reverse_key);
+    if (entry && reverse_entry) {
```

Comment:
i don't think this is correct, since if we've reached this part of the code, then we haven't found _either_ entries. If you look in `ct_process_packet`, you will see that `_ct_handle_new_connection`, and subsequently `_ct_handle_tcp_connection`, is called if either entry weren't found.